plugins/modules/cohesity_source.py

```diff
@@ -547,21 +548,31 @@ def main():
                     "msg"
                 ] = "Check Mode: Cohesity Protection Source is currently registered.  No changes"
             else:
-                check_mode_results[
-                    "msg"
-                ] = "Check Mode: Cohesity Protection Source is not currently registered.  This action would register the Protection Source."
+                # Check source is reachable.
+                endpoint = module.params.get("endpoint")
+                command = ["ping", "-c", "1", endpoint, "-w", "10"]
```

Comment:
Hmm, in some customers port 80 (ping) is not open. Probably check if there's agent status for registered physical sources.. for vmware ..the parent source status